go/worker/event/kafkaServer.go

```diff
@@ -0,0 +1,311 @@
+package event
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/open-lambda/open-lambda/go/common"
+	"github.com/open-lambda/open-lambda/go/worker/lambda"
+)
+
+
+// LambdaKafkaConsumer manages Kafka consumption for a specific lambda function
+type LambdaKafkaConsumer struct {
+	consumerName  string // Unique name for this consumer (may include index suffix)
+	lambdaName    string // Actual lambda function name for invocation
+	kafkaTrigger  *common.KafkaTrigger
+	client        *kgo.Client
+	lambdaManager *lambda.LambdaMgr // Reference to lambda manager for direct calls
+	stopChan      chan struct{}
+}
+
+// KafkaServer manages multiple lambda-specific Kafka consumers
+type KafkaServer struct {
+	lambdaConsumers map[string]*LambdaKafkaConsumer // lambdaName -> consumer
+	lambdaManager   *lambda.LambdaMgr          // Reference to lambda manager
+	mu              sync.RWMutex
```

Comment:
Is this just for protecting lambdaConsumers?  Need commenting about what a lock protects.  Let's just use a simple mutex too for simplicity (not RW).